cmd/vulcan-vulners/main.go

```diff
@@ -234,19 +252,35 @@ func analyzeReport(target string, nmapReport *gonmap.NmapRun) ([]report.Vulnerab
 			for _, cpe := range port.Service.CPEs {
 				logger.Debugf("CPE found: %v", cpe)
 				done = true
-
-				v, err := vulnerabilitiesByCPE(string(cpe))
+				f, err := findingByCPE(string(cpe))
 				if err != nil {
 					return nil, err
 				}
-
-				if v != nil {
-					v.Summary = fmt.Sprintf(v.Summary, port.Service.Product)
-					v.Description = fmt.Sprintf(v.Description, port.Service.Product)
-					v.Details = fmt.Sprintf("Found at port %d/%s", port.PortId, port.Protocol)
-
-					vulns = append(vulns, *v)
+				if f == nil {
+					continue
 				}
+				summary := fmt.Sprintf(vulnersVuln.Summary, port.Service.Product)
+				v, ok := uniqueVulns[summary]
+				if !ok {
+					v.Vuln = report.Vulnerability{
+						Summary:         summary,
+						Description:     fmt.Sprintf(vulnersVuln.Description, port.Service.Product),
+						Recommendations: vulnersVuln.Recommendations,
+					}
+					v.CPEs = map[string]struct{}{}
+					uniqueVulns[summary] = v
+				}
+				if _, ok := v.CPEs[string(cpe)]; !ok {
+					v.CPEs[string(cpe)] = struct{}{}
+					v.Vuln.Resources = append(v.Vuln.Resources, f.Resource)
+					uniqueVulns[summary] = v
+					if f.Score > v.Vuln.Score {
```

Comment:
I think it's possible to miss a higher Score from a different finding in the same port+cpe if this verification happens only when we are initialising the entry, no?

Reply:
I'm not sure because the the vulnerabilities are not host/port dependant. If you have findings for a CPE or Product+Version they will be the same independently from the host/port, and the score calculated for the first occurrence of the CPE/Product+Version should be the same than the rest.

Reply:
Yes I assumed what Julian said that's the reason I'm not updating the max score.

Reply:
Ok!